### ISSUE TITLE
History SQL queries aren't legal in some sqlite configurations - FreeBSD

### ISSUE THREAD

**Issue by toofar** [qutebrowser/qutebrowser#7596]
**Version info**:
This bug (double quotes around string literals in SQL queries) has been on the main branch since 2017: https://github.com/qutebrowser/qutebrowser/blob/bbb44f6/qutebrowser/browser/history.py#L222-L229

nimaje on IRC reported that qutebrowser wasn't starting on FreeBSD because of an error saying something about

    the 'qute://%' in 'not url like "qute://%"' in the history query should be a column

It is probably related to this recent change to the sqlite build in FreeBSD: https://cgit.freebsd.org/ports/commit/databases/sqlite3?id=47912ce2e613211aecf6ecdfc58681b7aca0df33

**Does the bug happen if you start with `--temp-basedir`?**:

It should do if you can reproduce it. :)

**Description**

TL;DR: we are using an optional feature of SQlite by mistake

In SQL double quotes are used around identifiers (table names, column names etc - they are required when the identifier contains a special character like space or period, or shadows a keyword, but are optional otherwise) and single quotes are always used around string literals (eg values being inserted or values to match against).

SQLite apparently allows using double quotes for quoting string literals for compatibility with MySQL, but it's discouraged because it's not compatible with other databases (like postgres and SQL Server): https://www.sqlite.org/quirks.html#double_quoted_string_literals_are_accepted

It's very unlikely that where would be any downsides to changing the SQL to use single quotes around string literals.

**How to reproduce**

This might be quite tricky since we use sqlite via QtSQL, here are some options I can think of:

1. use an up to date FreeBSD to test
2. patch qtbase/src/3rdparty/sqlite to disable DQS
3. configure Qt to use the system libsqlite and build you own one with DQS disabled

[The docs](https://www.sqlite.org/quirks.html#double_quoted_string_literals_are_accepted) mention some ways to disable DQS via the sqlite config API but that's only per connections and I don't think we have access to that API when accessing the DB via QtSQL (or python for that matter).

**Questions**

1. Can we help end users to mitigate this if they are on an affected platform with changing the source code? I couldn't see a way. We don't seem to have a way to change the option from Qt/python
2. Do we need to make a new stable release? Hmm, not sure. I don't think disabling this feature is a widespread trend, someone just thought it was a good idea in this case and didn't provide any explanation. The advice from the the mainterner to affected uses was to recompile sqlite. If they think that is enough for their userbase maybe it is: https://reviews.freebsd.org/D38774
3. Any other queries that need updating? I don't know. Could we run the test suite with any of the reproduction steps? Probably manually looking through any database related stuff would be easier, there can't be that many

**Comment by The-Compiler** [comment]
I'm getting lots of crash reports about this. I suppose we'll need a quick v2.5.4 with this fixed, maybe as a source-only release.
